source/run-command/run-command.rst

```diff
@@ -187,7 +193,157 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
+A driver can expect that the result from running this command will return a document with a ``cursor`` field and MUST provide the caller with a language native abstraction to continue iterating the results from the server.
+
+High level RunCursorCommand steps:
+
+* If no session is provided by the caller, create a ClientSession
+* Create a local Cursor instance
+* Run the cursor creating command provided by the caller
+* Store the ``firstBatch``, ``ns``, and ``id`` from the server's response as well as which server the operation was executed on
+* When the current batch has been fully iterated, using the same server the initial command was executed on execute a ``getMore``
+* Store the results of the getMore's ``nextBatch`` and update the cursor's ``id``
+* Continue executing ``getMore`` commands until the cursor is exhausted
+
+Driver Sessions
+"""""""""""""""
+
+A driver MUST create an implicit ClientSession if none is provided and it MUST be attached for the duration of the cursor's lifetime.
+All ``getMore`` commands constructed for this cursor MUST send the same ``lsid`` used on the initial command.
+A cursor is exhausted when the server reports a cursor is equal to zero.
+When the cursor is exhausted the client session MUST be ended and the server session returned to the pool as early as possible rather than waiting for a caller to completely iterate the final batch.
+
+Server Selection
+""""""""""""""""
+
+RunCursorCommand MUST support a ``readPreference`` option that MUST be used to determine server selection.
+The selected server MUST be used for subsequent ``getMore`` commands.
+
+Load Balancers
+""""""""""""""
+
+When in ``loadBalanced`` mode, a driver MUST pin the connection the initial operation was executed on and reuse it for subsequent ``getMore`` operations.
+
+* See Load Balancer's section on `Behaviour With Cursors <https://github.com/mongodb/specifications/blob/master/source/load-balancers/load-balancers.rst#behaviour-with-cursors>`_
+
+Iterating the Cursor
+""""""""""""""""""""
+
+Drivers MUST provide an API, typically, a method named ``next()``, that returns one document per invocation.
+If the cursor's batch is empty and the cursor id is nonzero, the driver MUST perform a ``getMore`` operation.
+
+Executing GetMores
+""""""""""""""""""
+
+The cursor API returned to the caller MUST offer a way to configure ``batchSize``, ``maxTimeMS``, and ``comment`` that are sent on the ``getMore`` command.
+
+.. code:: typescript
+
+  interface GetMore {
+    /** Set to the nonzero cursor id */
+    getMore: int64;
+    /** Set to the namespace returned by the initial command response */
+    collection: string;
+    /**
+     * User configurable document count for the batch returned for this getMore.
+     * Only attached to command document if nonzero.
+     */
+    batchSize?: int23;
```

Comment:
```suggestion
    batchSize?: int32;
```

I shudder to think of the complexities of developing for a 23-bit architecture.

Also, TIL [45-bit architectures](https://en.wikipedia.org/wiki/45-bit_computing) exist(ed).

Reply:
I was planning on amending the BSON spec after this /s